pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -104,14 +105,22 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundHTTPRouteConfig(
 	return r
 }
 
+// IPv6 addresses are enclosed within square brackets
+func ipv6Compliant(host string) string {
+	if net.ParseIP(host) != nil && strings.Contains(host, ":") {
```

Comment:
ParseIP can be removed

Reply:
added as additional check to verify if it is just ip address although not needed as ':' is not valid in dns name. If address already includes brackets, this will prevent adding it again.  shall I keep it or remove?

Reply:
are we certain this can never get inputs like `foo.com:80`?

Reply:
It is certain now, as far as  no misuse later

Reply:
@hzxuzhonghu what about for this? - If address already includes brackets, this will prevent adding it again.

Reply:
Is it valid for host to include `[]`? So this depends on this.

Reply:
Not as per current definition of host in virtualservice spec, which is wildcard, ip or hostname. But in general, valid in URI/domain context.